include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -41,9 +37,23 @@ struct Exception : std::exception {
   OrtErrorCode code_;
 };
 
-// This Macro is to make it easy to generate overloaded methods for all of the various OrtRelease* functions for every Ort* type
+// This is used internally by the C++ API. This class holds the global variable that points to the OrtApi, it's in a template so that we can define a global variable in a header and make
+// it transparent to the users of the API.
+template <typename T>
+struct Global {
+  static const OrtApi& api_;
+};
+
+template <typename T>
+const OrtApi& Global<T>::api_ = *OrtGetApiBase()->GetApi(ORT_API_VERSION);
```

Comment:
>const OrtApi& Global<T>::api_ [](start = 0, length = 29)

Can this be made a function local static?

Reply:
No, it has to be outside of the template to work as a global variable.